app/Services/IssueService.php

```diff
@@ -19,12 +19,22 @@ class IssueService
      *
      * @return array<Issue>
      */
-    public function getAll(): array
+    public function getAll(String $sort = null): array
     {
-        return collect(config('repos.repos'))
-            ->flatMap(fn (array $repo): array => $this->getIssuesForRepo($repo))
-            ->shuffle()
-            ->all();
+        if ($sort) {
```

Comment:
I think we might be able to clean up this code a little bit using this type of approach:

```php
$allRepos = collect(config('repos.repos'))
    ->flatMap(fn (array $repo): array => $this->getIssuesForRepo($repo))
    ->when(
        $sort,
        fn (Collection $collection): Collection => $collection->sortBy($sort),
        fn (Collection $collection): Collection => $collection->shuffle()
    )
    ->all();
```

I've not run this code, but I think it could possibly work

Reply:
Agree this is much tidier, I need to improve my knowlege of collections!